lib/libxdp/libxdp.c

```diff
@@ -2607,6 +2614,14 @@ bool xdp_multiprog__is_legacy(const struct xdp_multiprog *mp)
 	return !!(mp && mp->is_legacy);
 }
 
+int xdp_multiprog__program_count(const struct xdp_multiprog *mp)
+{
+	if (!mp)
+		return -EINVAL;
+
+	return mp->num_links;
```

Comment:
Any reason why using mp->num_links here, and above mp->config.num_progs_enabled?

Reply:
If this is ok with @tohojo I have no further comments and you can go ahead with the merge.

Reply:
I used `mp->config.num_progs_enabled` because it returns the total amount of programs to be loaded, compared to `mp->num_links` returning the amount of currently loaded programs. So since `xdp_multiprog__link_prog` is called while programs are still being loaded `mp->num_links` would return just the number of already loaded programs, which would always be 1 once.

Reply:
Thanks this is now clear.